### PR TITLE
sql: CONNECTION option unification

### PR DESCRIPTION
**Pull request by sploiselle**
In support of the impending `ALTER CONNECTION` PR, we want to unify all of the connection options into one enum--this will let users specify that they want to alter some property of a connection without specifying its type. We can then check the type during planning to ensure that everything matches. [Nikhil, Moritz, and I all agreed](https://github.com/MaterializeInc/materialize/pull/21109#discussion_r1290517235) that this approach is the right tradeoff for UX vs. dev ergonomics, so am looking for a review of the submitted code and not the proposed design.

### Motivation

This PR refactors existing code.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by benesch — COMMENTED** [review]
I remain in favor of this design. Deferring actual review to Joe, but nothing looked awry on a quick skim. I'm so, so sorry that you had to assemble this PR though. What a mountain of code to move around!

**Review by jkosh44 — APPROVED** [review]
LGTM

File: src/sql-parser/src/ast/defs/ddl.rs [hunk 1]
```diff
@@ -600,258 +600,81 @@ impl<T: AstInfo> AstDisplay for DbzTxMetadataOption<T> {
 }
 impl_display_t!(DbzTxMetadataOption);
 
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub enum KafkaConnectionOptionName {
+// All connection options are bundled together to allow us to parse `ALTER
+// CONNECTION` without specifying the type of connection we're altering. Il faut
+// souffrir pour être belle.
```
[the remark this answers is not in the document]
Comment (sploiselle) [reply src/sql-parser/src/ast/defs/ddl.rs:605]: https://commons.wikimedia.org/wiki/File:Il_Faut_Souffrir_Pour_%C3%8Atre_Belle.jpg

File: src/sql-parser/src/ast/defs/statement.rs [hunk 2]
```diff
@@ -618,12 +619,13 @@ impl<T: AstInfo> AstDisplay for KafkaBrokerAwsPrivatelink<T> {
 }
 impl_display_t!(KafkaBrokerAwsPrivatelink);
 
-/// `CREATE CONNECTION`
+/// `CREATE CONNECTION` refactor WIP
```
[the remark this answers is not in the document]
Comment (sploiselle) [reply src/sql-parser/src/ast/defs/statement.rs:622]: Ah, I'll make sure that's struck from the ALTER CONNECTION PR--good catch.

**Pull request merged**